code/controllers/subsystems/persistence.dm

```diff
@@ -0,0 +1,212 @@
+		database_clean_entries()
+
+		// Retrieve all persistent data that is not expired
+		var/list/persistent_data = database_get_active_entries()
+
+		// Instantiate all remaining entries based of their type
+		// Assign persistence related vars found in /obj, apply content and add to live tracking list.
+		for (var/data in persistent_data)
+			var/typepath = text2path(data["type"])
+			if (!ispath(typepath)) // Type checking
+				continue
+			var/obj/instance = new typepath()
+			instance.persistence_track_id = data["id"]
+			instance.persistence_author_ckey = data["author_ckey"]
+			instance.persistence_apply_content(data["content"], data["x"], data["y"], data["y"])
+			register_track(instance)
+
+		return SS_INIT_SUCCESS
+
+/**
+ * Shutdown of the persistence subsystem. Adds new persistent objects, removes no longer existing persistent objects and updates changed persistent objects in the database.
+ */
+/datum/controller/subsystem/persistence/Shutdown()
+	// Subsystem shutdown:
+	// Create new persistent records for objects that have been created in the round
+	// Update tracked objects that have an ID (already existing from previous rounds) if they have changed
+	// Delete persistent records that no longer exist in the registry (removed during the round)
+
+	var/list/track_lookup = list()
+
+	// Iterate through the register to sort tracks with no ID and tracks that may need an update (tracks with ID)
+	// We are using a dictionary look-up to find no longer existing records by comparing them with a live dataset later on
+	for (var/obj/track in GLOB.persistence_register)
+		if(track.persistence_track_id > 0) // (0 is the default tracking ID value)
+			// Tracked object has an ID, add it to lookup
+			track_lookup[track.persistence_track_id] = track
+		else
+			// Tracked object has no ID, create a new persistent record for it
+			database_add_entry(track)
+
+	for(var/record in database_get_active_entries())
+		// Find removed objects by looking them up using the live dataset
+		var/obj/track = track_lookup[record["id"]]
+
+		if (track)
+			// The record still exists as an active track, check if it may need an update
+			var/changed = FALSE
+			if (track.persistence_author_ckey != record["author_ckey"])
+				changed = TRUE
+			else if (track.persistence_get_content() != record["content"])
+				changed = TRUE
+			else if (track.x != record["x"])
+				changed = TRUE
+			else if (track.y != record["y"])
+				changed = TRUE
+			else if (track.z != record["z"])
+				changed = TRUE
+			if (changed == TRUE)
+				database_update_entry(track)
+		else
+			// There was no match in the lookup meaning the object got removed this round
+			// We delete persistent data by setting it's expiration date to now, actual deletion has more logic and is handled in cleanup during init.
+			database_expire_entry(record["id"])
```

Comment:
Should there be a way to prevent entries from expiring?
(i.e. if we want to pin something to a notice board due to a ongoing lore event?)